### PR TITLE
test(#33): add test for email preference controller

### PR DESCRIPTION
**Pull request by sathia27**
add test for email preference controller which resolves issue #33

### CODE REVIEW

File: spec/controllers/api/v1/email_preferences_controller_spec.rb [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+  describe "PUT #update" do
+    context "when user has no email preference" do
+      it "should return success" do
+        put :update, format: :json, params: {email_preference: {news: true}}
+
+        assert_response :success
+      end
+
+      it "should create and return new email preference" do
+        email_preference = {"news" => true, "likes" => false, "messages" => false}
```
Comment (AlexAvlonitis) [inline spec/controllers/api/v1/email_preferences_controller_spec.rb]: Could you create an email factory instead, as it's an actual model.
https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#using-factories

you can find other factories in this folder spec/factories

File: spec/controllers/api/v1/email_preferences_controller_spec.rb [hunk 2]
```diff
@@ -1,5 +1,54 @@
...
+      end
+    end
+
+    context "when user has aleady email preference" do
+      it "should update and return updated email preference" do
+        email_preference = @user.build_email_preference
+        email_preference.likes = true
+        email_preference.messages = true
+        email_preference.news = false
+        email_preference.save
```
Comment (AlexAvlonitis) [inline spec/controllers/api/v1/email_preferences_controller_spec.rb]: Could you create an email factory instead as the previous comment. 
https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#using-factories

File: spec/controllers/api/v1/email_preferences_controller_spec.rb [hunk 3]
```diff
@@ -1,5 +1,54 @@
...
+        email_preference = {"news" => false, "likes" => true, "messages" => true}
+        put :update, format: :json, params: {email_preference: email_preference}
+
+        parsed_body = JSON.parse(response.body)
+        expect(parsed_body['email_preference']).to match(email_preference)
+      end
+    end
+
+    context "when update failed" do
+      it "should return :unprocessable_entity with errors" do
```
Comment (AlexAvlonitis) [inline spec/controllers/api/v1/email_preferences_controller_spec.rb]: Could you remove all the shoulds please. and instead 'returns ...`
http://www.betterspecs.org/#should

**Comment by AlexAvlonitis** [comment]
great job @sathia27 added few minor comments

**Comment by sathia27** [comment]
@AlexAvlonitis Thanks for your comments. Please review the new changes.

**Comment by AlexAvlonitis** [comment]
excellent @sathia27

**Pull request merged**